src/extensions/askUserForParam.ts

```diff
@@ -116,6 +129,81 @@ export async function askForParam(paramSpec: Param): Promise<string> {
   return response;
 }
 
+async function promptSecret(
```

Comment:
Somewhere in this code path, we should ensure(https://github.com/firebase/firebase-tools/blob/master/src/ensureApiEnabled.ts#L116) that the secret manager API is enabled,
 and requirePermissions(https://github.com/firebase/firebase-tools/blob/master/src/requirePermissions.ts#L18) to check that the user has the required IAM roles to do this. Ideally, we only should do these if there is a secret & only once even if we have multiple secrets.
 
 Making these calls before hand ensures that we provide clear, actionable error messages when the API isn't enabled/IAM roles are missing.

Reply:
Good catch! I totally forgot. I had to also tweak the "billing required" logic for this (secrets always require billing) and I ensure secretmanager API is enabled there too (if secrets are used).